### DISCUSSION TITLE
Lots of compilation errors on Kubuntu 20.04

### DISCUSSION THREAD

**Discussion by Arcky87** [iraf-community/iraf#151] [Q&A] [answered]
Good day,

I faced some problems building latest IRAF release from the source on Kubuntu 20.04. I made all the steps according to the installation instruction, everything seems to work fine until I run $ make sysgen 2>&1 | tee build.log and get multiple errors like this:

```
In file included from irafpath.c:11:
/usr/local/iraf/include/iraf.h:1:1: error: expected identifier or ‘(’ before ‘.’ token
    1 | ../unix/hlib/libc/iraf.h
      | ^
In file included from prwait.c:11:
/usr/local/iraf/include/iraf.h:1:1: error: expected identifier or ‘(’ before ‘.’ token
    1 | ../unix/hlib/libc/iraf.h
      | ^
prwait.c:31:11: error: ‘MAXPROCS’ undeclared here (not in a function)
   31 | } prtable[MAXPROCS];


```
Complete build.log is attached.
[build.log](https://github.com/iraf-community/iraf/files/5824579/build.log)

I believe these errors probably occur due to wrong gcc definitions but need some help to solve them. Thanks!

**Comment by olebole** [comment]
Why don't you just use the IRAF that comes with kubuntu? `sudo apt install iraf` gives it in a minute.
In you case it seems that you have an old IRAF installation in `/usr/local/iraf`, and your `$iraf` environment variable points there. All the IRAF environment vars should be cleaned before starting to compile.

**Comment by Arcky87** [comment]
> Why don't you just use the IRAF that comes with kubuntu?

Will give it a try, thank you. Does it still support gui packages like spectool?

 > In you case it seems that you have an old IRAF installation in `/usr/local/iraf`
This is exactly the folder i wanna install iraf to.

**Comment by olebole** [comment, marked as answer]
The IRAF in Ubuntu is built from the same sources and therefore has the same features. GUI is rather a matter of [x11iraf](https://github.com/iraf-community/x11iraf), and there the versions included in Debian and Ubuntu have [all non-free stuff removed](https://github.com/iraf-community/x11iraf/pull/20), so you should test whether this includes the features you need.

If you want to have it installed in `/usr/local`, you should unpack the tarball there, removing the old installation before. IRAF is there different from other programs that it does an "in-place" installation only.

**Comment by Arcky87** [comment]
Thanks for the help. I'm closing the issue
